code/modules/mining/boulder_processing/boulder.dm

```diff
@@ -0,0 +1,271 @@
+			if(/datum/material/titanium)
+				cracked_ore = new /obj/item/stack/ore/titanium(drop_location(), quantity)
+			if(/datum/material/uranium)
+				cracked_ore = new /obj/item/stack/ore/uranium(drop_location(), quantity)
+		SSblackbox.record_feedback("tally", "ore_mined", quantity, cracked_ore)
+
+/obj/item/boulder/proc/can_get_processed()
+	if(COOLDOWN_FINISHED(src, processing_cooldown))
+		return TRUE
+	return FALSE
+
+/obj/item/boulder/proc/reset_processing_cooldown()
+	COOLDOWN_START(src, processing_cooldown, 2 SECONDS)
+
+/**
+ * Moves boulder contents to the drop location, and then deletes the boulder.
+ */
+/obj/item/boulder/proc/break_apart()
+	var/list/quips = list("Clang!", "Crack!", "Bang!", "Clunk!", "Clank!")
+	for(var/obj/item/content as anything in contents)
+		content.forceMove(get_turf(src))
+		visible_message(span_notice("[pick(quips)] Something falls out of \the [src]!"))
+		playsound(loc, 'sound/effects/picaxe1.ogg', 60, FALSE)
+	qdel(src)
+
+/**
+ * This is called when a boulder is spawned from a vent, and is used to set the boulder's icon as well as durability.
+ * We also set our boulder_size variable, which is used for inheiriting the icon_state later on if processed.
+ */
+/obj/item/boulder/proc/flavor_boulder(obj/structure/ore_vent/parent_vent, size = BOULDER_SIZE_SMALL, is_artifact = FALSE)
+	var/durability_min = size
+	var/durability_max = size + BOULDER_SIZE_SMALL
+	if(parent_vent)
+		durability_min = parent_vent.boulder_size
+		durability_max = parent_vent.boulder_size + BOULDER_SIZE_SMALL
+	durability = rand(durability_min, durability_max) //randomize durability a bit for some flavor.
+	boulder_size = size
+	if(parent_vent)
+		boulder_size = parent_vent.boulder_size
+		boulder_string = parent_vent.boulder_icon_state
+	update_appearance(UPDATE_ICON_STATE)
+
+/**
+ * Unique proc for gulag-style boulders, which adds a random amount of minerals to the boulder.
+ */
+/obj/item/boulder/proc/add_gulag_minerals(list_to_pick)
+	var/datum/material/new_material = pick_weight(gulag_minerals)
+	var/list/new_mats = list()
+	new_mats += new_material
+	new_mats[new_material] = SHEET_MATERIAL_AMOUNT * rand(1,3) //We only want a few sheets of material in the gulag boulders
+	set_custom_materials(new_mats)
+
+/obj/item/boulder/artifact
+	name = "artifact boulder"
+	desc = "This boulder is brimming with strange energy. Cracking it open could contain something unusual for science."
+	icon_state = "boulder_artifact"
+	/// This is the type of item that will be inside the boulder. Default is a strange object.
+	var/artifact_type = /obj/item/relic
+	/// References to the relic inside the boulder, if any.
+	var/obj/item/artifact_inside
+
+/obj/item/boulder/artifact/Initialize(mapload)
+	. = ..()
+	artifact_inside = new artifact_type(src) /// This could be poggers for archaeology in the future.
+
+/obj/item/boulder/artifact/Destroy(force)
+	artifact_inside = null
```

Comment:
```suggestion
	QDEL_NULL(artifact_inside)
```
In case `convert_to_ore()` does not get called and the artifact is still inside we should make sure to delete it